lib/assessment/rubric_tester.py

```diff
@@ -151,53 +189,62 @@ def grade_student_work(prompt, rubric, student_file, examples, options):
-
-    accuracy_by_criteria, overall_accuracy = compute_accuracy(expected_grades, actual_grades, options.passing_grades)
-    report = Report()
-    report.generate_html_output(
-        output_file, prompt, rubric, overall_accuracy, actual_grades, expected_grades, options.passing_grades, accuracy_by_criteria, errors, command_line
-    )
-    logging.info(f"main finished in {int(time.time() - main_start_time)} seconds")
 
-    os.system(f"open {output_file}")
+    for lesson in options.lesson_names:
+        prefix = os.path.join(base_dir, lesson)
+
+        # download lesson files
+        if not os.path.exists(prefix) or options.download:
+            gdown.download_folder(id=LESSONS[lesson], output=prefix)
+
+        # read in lesson files, validate them
+        prompt, standard_rubric = read_inputs(prompt_file, standard_rubric_file, prefix)
+        student_files = get_student_files(options.max_num_students, prefix, student_ids=options.student_ids)
+        expected_grades = get_expected_grades(expected_grades_file, prefix)
+        examples = get_examples(prefix)
+
+        validate_rubrics(expected_grades, standard_rubric)
+        validate_students(student_files, expected_grades)
+        rubric = standard_rubric
+
+        # set up output and cache directories
+        output_file = os.path.join(prefix, output_dir_name, options.output_filename)
+        os.makedirs(os.path.dirname(output_file), exist_ok=True)
+        os.makedirs(os.path.join(prefix, cache_dir_name), exist_ok=True)
+        if not options.use_cached:
+            for file in glob.glob(f'{os.path.join(prefix, cache_dir_name)}/*'):
+                os.remove(file)
+
+        # call grade function to either call openAI or read from cache
+        with concurrent.futures.ThreadPoolExecutor(max_workers=7) as executor:
+            actual_grades = list(executor.map(lambda student_file: grade_student_work(prompt, rubric, student_file, examples, options, prefix), student_files))
+
+        errors = [student_id for student_id, grades in actual_grades if not grades]
+        # actual_grades contains metadata and data (grades), we care about the data key
+        actual_grades = {student_id: grades['data'] for student_id, grades in actual_grades if grades}
+
+        # calculate accuracy and generate report
+        accuracy_by_criteria, overall_accuracy, confusion_by_criteria, overall_confusion, grade_names = compute_accuracy(expected_grades, actual_grades, options.passing_grades)
+        report = Report()
+        report.generate_html_output(
+            output_file, prompt, rubric, overall_accuracy, actual_grades, expected_grades, options.passing_grades, accuracy_by_criteria, errors, command_line, confusion_by_criteria, overall_confusion, grade_names, prefix=prefix
+        )
+        logging.info(f"main finished in {int(time.time() - main_start_time)} seconds")
+
+        os.system(f"open {output_file}")
```

Comment:
one more thought: could this file be moved to the bin directory? I could be missing something, but it seems like it is more of a binary than a library